### PR TITLE
flint: 2.5.2 -> 2.7.1

### PR DESCRIPTION
**Pull request by TethysSvensson**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/flint/default.nix [hunk 1]
```diff
@@ -41,14 +41,6 @@ stdenv.mkDerivation rec {
   # issues with ntl -- https://github.com/wbhart/flint2/issues/487
   NIX_CXXSTDLIB_COMPILE = "-std=c++11";
 
-  patches = [
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/flint/default.nix:45]: Please remove fetchpatch from inputs.

Comment (TethysSvensson) [reply pkgs/development/libraries/flint/default.nix:45]: Done

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118730` at 962cac65 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>polymake</li>
  </ul>
</details>
<details>
  <summary>7 packages built successfully:</summary>
  <ul>
    <li>arb</li>
    <li>flint</li>
    <li>pynac</li>
    <li>python38Packages.symengine</li>
    <li>python39Packages.symengine</li>
    <li>singular</li>
    <li>symengine</li>
  </ul>
</details>

**Comment by TethysSvensson** [comment]
@SuperSandro2000 Would you mind taking another look?

**Pull request merged**